00_Base/src/mapper/cdr.mapper.ts

```diff
@@ -1,62 +1,53 @@
+import { SequelizeTariffRepository, Tariff, Transaction } from '@citrineos/data';
 import { SessionMapper } from './session.mapper';
-import { OcpiLogger } from '../util/logger';
 import { CdrLocation } from '../model/CdrLocation';
-import { TransactionEventRequest } from '@citrineos/base';
 import { Price } from '../model/Price';
 import { Tariff as OcpiTariff } from '../model/Tariff';
 import { SignedData } from '../model/SignedData';
 import { LocationDTO } from '../model/DTO/LocationDTO';
 import { TariffsService } from '../services/tariffs.service';
+import { BaseTransactionMapper } from './BaseTransactionMapper';
+import { ILogObj, Logger } from 'tslog';
+import { OcpiLocationRepository } from '../repository/OcpiLocationRepository';
+import { TokensRepository } from '../repository/TokensRepository';
+import { LocationsService } from '../services/locations.service';
 
 @Service()
-export class CdrMapper {
+export class CdrMapper extends BaseTransactionMapper {
   constructor(
+    protected logger: Logger<ILogObj>,
+    protected locationsService: LocationsService,
+    protected ocpiLocationsRepository: OcpiLocationRepository,
+    protected tokensRepository: TokensRepository,
+    protected tariffRepository: SequelizeTariffRepository,
+    protected tariffsService: TariffsService,
     readonly sessionMapper: SessionMapper,
-    readonly logger: OcpiLogger,
-    readonly locationRepository: SequelizeLocationRepository,
-    readonly tariffRepository: SequelizeTariffRepository,
-    readonly tariffsService: TariffsService,
-  ) {}
+  ) {
+    super(logger, locationsService, ocpiLocationsRepository, tokensRepository, tariffRepository, tariffsService);}
 
   public async mapTransactionsToCdrs(
-    transactions: Transaction[],
-    fromCountryCode?: string,
-    fromPartyId?: string,
-    toCountryCode?: string,
-    toPartyId?: string,
+    transactions: Transaction[]
   ): Promise<Cdr[]> {
     try {
       const validTransactions = this.getCompletedTransactions(transactions);
 
       const sessions = await this.mapTransactionsToSessions(
         validTransactions,
-        fromCountryCode,
-        fromPartyId,
-        toCountryCode,
-        toPartyId,
       );
 
-      const transactionIdToTariffMap: Map<string, Tariff> =
-        await this.getTariffsForTransactions(validTransactions);
+      const [transactionIdToTariffMap, transactionIdToLocationMap] = await
+        Promise.all([this.getTariffsForTransactions(validTransactions), this.getLocationDTOsForTransactions(transactions)]);
       const transactionIdToOcpiTariffMap: Map<string, OcpiTariff> =
         await this.getOcpiTariffsForTransactions(
           sessions,
-          transactionIdToTariffMap,
-          toCountryCode,
-          toPartyId,
+          transactionIdToTariffMap
         );
       return await this.mapSessionsToCDRs(
         sessions,
+        transactionIdToLocationMap,
         transactionIdToTariffMap,
         transactionIdToOcpiTariffMap,
```

Comment:
For simplicity sake, I would love if this could be as simple as "Build CDRs given OCPI Locations and OCPI Tariffs".

But instead, we pass in both Core Tariffs and OCPI Tariffs.

Reply:
Makes sense, will track with ticket